### PR TITLE
Avoid spuriously adding Geometry capability for vert, tesc, tese

### LINKED ISSUE #2461: Use of gl_Layer in tessellation evaluation shader adds Geometry capability
This was originally reported in  https://github.com/google/shaderc/issues/1112 from a use case with Molten-VK.
cc: @vladimir-nazarenko

I have a reproducer:

```
#version 450

#extension GL_ARB_shader_viewport_layer_array : require

layout(triangles) in;
void main() {
  gl_Layer = 1;
}
```

Compiling:
glslangValidator -V t.tese -o t.spv

Produces:
```
; SPIR-V
; Version: 1.0
; Generator: Khronos Glslang Reference Front End; 10
; Bound: 10
; Schema: 0
               OpCapability Geometry
               OpCapability Tessellation
               OpCapability ShaderViewportIndexLayerEXT
               OpExtension "SPV_EXT_shader_viewport_index_layer"
          %1 = OpExtInstImport "GLSL.std.450"
               OpMemoryModel Logical GLSL450
               OpEntryPoint TessellationEvaluation %main "main" %gl_Layer
               OpExecutionMode %main Triangles
               OpExecutionMode %main SpacingEqual
               OpExecutionMode %main VertexOrderCcw
               OpSource GLSL 450
               OpSourceExtension "GL_ARB_shader_viewport_layer_array"
               OpName %main "main"
               OpName %gl_Layer "gl_Layer"
               OpDecorate %gl_Layer BuiltIn Layer
       %void = OpTypeVoid
          %3 = OpTypeFunction %void
        %int = OpTypeInt 32 1
%_ptr_Output_int = OpTypePointer Output %int
   %gl_Layer = OpVariable %_ptr_Output_int Output
      %int_1 = OpConstant %int 1
       %main = OpFunction %void None %3
          %5 = OpLabel
               OpStore %gl_Layer %int_1
               OpReturn
               OpFunctionEnd
```

### PR DESCRIPTION
**Pull request by dneto0**
Use of gl_Layer and gl_ViewportIndex in tessellation and vertex
shaders should not trigger the addition of the Geometry capability.

Fixes #2461

Added tests for use of gl_Layer and gl_ViewportIndex in a tessellation
evaluation shader.

Several tests for NVIDIA features for tessellation, vertex, or mesh
shaders now lose the Geometry or MultiViewport capabilities.
This is ok because the functionality is already covered by
the ShaderViewportIndexLayerNV capability.

The spv.meshShaderPerViewBuiltins.mesh test now fails validation
because the validator does not know that PrimitiveId (and possibly
other) builtins are enabled by the MeshShadingNV capability.
I filed https://github.com/KhronosGroup/SPIRV-Headers/issues/179 to
fix the grammar upstream.

### CODE REVIEW

**Comment by dneto0** [comment]
cc: @dgkoch @jeffbolznv

**Comment by dneto0** [comment]
The validation error should go away once [this change](https://github.com/KhronosGroup/SPIRV-Headers/pull/180) to SPIRV-Headers is included by Glslang's known_good.json

File: Test/baseResults/spv.meshShaderPerViewBuiltins.mesh.out [hunk 1]
```diff
@@ -1,9 +1,9 @@
 spv.meshShaderPerViewBuiltins.mesh
+Validation failed
```
Comment (jeffbolznv) [inline Test/baseResults/spv.meshShaderPerViewBuiltins.mesh.out:2]: This is a spirv-val bug, right? SPV_NV_mesh_shader says MeshShadingNV is sufficient here.

Comment (jeffbolznv) [reply Test/baseResults/spv.meshShaderPerViewBuiltins.mesh.out:2]: I should read the comments first next time, they already said this.

**Review by jeffbolznv — APPROVED** [review]

**Pull request merged**